nexus_constructor/stream_fields_widget.py

```diff
@@ -0,0 +1,243 @@
+
+        self.layout().addWidget(self.source_label, 5, 0)
+        self.layout().addWidget(self.source_line_edit, 5, 1)
+
+        self.layout().addWidget(self.hs00_unimplemented_label, 6, 0, 1, 2)
+
+        # Spans both rows
+        self.layout().addWidget(self.show_f142_advanced_options_button, 7, 0, 1, 2)
+        self.layout().addWidget(self.f142_advanced_group_box, 8, 0, 1, 2)
+
+        self._schema_type_changed(self.schema_combo.currentText())
+
+    def _show_advanced_options(self):
+        self.f142_advanced_group_box.setVisible(
+            not self.f142_advanced_group_box.isVisible()
+        )
+
+    def _show_array_size(self, show: bool):
+        self.array_size_spinbox.setVisible(show)
+        self.array_size_label.setVisible(show)
+
+    def _schema_type_changed(self, schema: str):
+        self.parent().setWindowTitle(f"Editing {schema} stream field")
+        self.hs00_unimplemented_label.setVisible(False)
+        self.f142_advanced_group_box.setVisible(False)
+        self.show_f142_advanced_options_button.setVisible(False)
+        if schema == "f142":
+            self._set_edits_visible(True, True)
+            self.show_f142_advanced_options_button.setVisible(True)
+            self.f142_advanced_group_box.setVisible(False)
+        elif schema == "ev42":
+            self._set_edits_visible(False, False)
+        elif schema == "hs00":
+            self._set_edits_visible(True, False)
+            self.hs00_unimplemented_label.setVisible(True)
+
+        elif schema == "ns10":
+            self._set_edits_visible(True, False, "nicos/<device>/<parameter>")
+
+    def _set_edits_visible(self, source: bool, type: bool, source_hint=None):
+        self.source_label.setVisible(source)
+        self.source_line_edit.setVisible(source)
+        self.type_label.setVisible(type)
+        self.type_combo.setVisible(type)
+        self.array_radio.setVisible(type)
+        self.scalar_radio.setVisible(type)
+        if source_hint:
+            self.source_line_edit.setPlaceholderText(source_hint)
+        else:
+            self.source_line_edit.setPlaceholderText("")
+
+    def get_stream_group(self) -> h5py.Group:
+        string_dtype = h5py.special_dtype(vlen=str)
+        temp_file = h5py.File(
+            name=str(uuid.uuid4()), driver="core", backing_store=False
+        )
+        group = temp_file.create_group("children")
+        group.create_dataset(name="type", dtype=string_dtype, data="stream")
+        stream_group = group.create_group(self.parent().parent().name)
+        stream_group.create_dataset(
+            name="topic", dtype=string_dtype, data=self.topic_line_edit.text()
+        )
+        stream_group.create_dataset(
+            name="writer_module",
+            dtype=string_dtype,
+            data=self.schema_combo.currentText(),
+        )
+
+        schema = self.schema_combo.currentText()
+
+        if schema == "f142":
+            stream_group.create_dataset(
```

Comment:
Extracting the contents of the `if` to a separate method would help keep things neat and readable here.